### DISCUSSION TITLE
m3u8 support

### DISCUSSION THREAD

**Discussion by NoOneNook** [Zibbp/ganymede#73] [Ideas]
Hi!
Have you ever considered adding support for converting mp4 to m3u8 and also adding support for playing m3u8 video?

I like m3u8 better than mp4 in the Twitch vod context, for some people it can take a very long time to load such large mp4 files. m3u8 solves this problem.

Ganymede is an incredibly well-executed project!

**Comment by Zibbp** [comment]
Hi,
I have not put any thought into converting to HLS playlists, or enabling an option to do so. A requirement I have for the saved files it to keep it as simple as possible which is why I went with mp4. I know over remote connections it can take a while to initially load the video file. Using HLS playlists would significantly decrease the initial load time.

I will take a look at possibly implementing an optional mp4 to HLS conversion job at the end of the video queue jobs. The HLS playlist would replace the mp4 file as to not have duplicate large video files.

**Comment by Zibbp** [comment]
@NoOneNook  I have released [v1.1.5](https://github.com/Zibbp/ganymede/releases/tag/v1.1.5) which includes converting the mp4 to a HLS playlist. More information about this can be found on the release page. If you have any questions or issues let me know.

**Comment by NoOneNook** [comment]
Hi!
Thank you for the good feedback.

This is very good news and I look forward to testing it in the future :D
